tests/structs/src/debug_derive.c

```diff
@@ -0,0 +1,29 @@
+//! derive:Debug
+
+#include <stdarg.h>
+
+typedef struct {
+    int a;
+} Debuggable1;
+
+typedef struct {
+    va_list v;
+} Debuggable2;
+
+Debuggable1 *kDebuggable1;
+Debuggable2 *kDebuggable2;
+
+// A struct containing a union cannot derive Debug, so make
+// sure we don't generate a #[derive] for it
+typedef struct {
+    struct {
+        struct {
+            union {
+                int d;
+                float e;
+            } c;
+        } b;
+    } a;
+} NotDebuggable1;
```

Comment:
You can test this is not `Debug` using [`static_assertions::assert_not_impl_any!`](https://docs.rs/static_assertions/1.1.0/static_assertions/macro.assert_not_impl_any.html).  There's also a similar `assert_impl_one!` you can use to do the check you're already doing, that a type is `Debug`.

Reply:
That way you also don't need to generate a value of the types, just the type itself.